src/sequence/SequenceDecoder.cpp

```diff
@@ -348,8 +347,10 @@ void SequenceDecoder::Decode(Instance *instance, Parts *parts,
 
     // Recover the best path in the original second-order model.
     RecoverBestPath(transformed_best_path, &best_path);
-  } else {
+  } else if (pipe_->GetSequenceOptions()->markov_order() == 1) {
     value = RunViterbi(node_scores, edge_scores, &best_path);
+  }else{
```

Comment:
Great that you implemented this! I think I had never tested 0-th Markov order before :-)

But, please fix the missing spaces "}else{" -> "} else {"